### PR TITLE
tests: improve test case about raft learner

### PR DESCRIPTION
**Pull request by hicqu**
## What have you changed? (mandatory)
Improve a test case about remove raft peer and then add it back.
It's needed because data is deleted asynchronous after the peer is removed. So if we call `must_get_equal` after add it back, the inner sleep time maybe not enough.

### CODE REVIEW

File: tests/raftstore_cases/test_conf_change.rs [hunk 1]
```diff
@@ -612,7 +612,9 @@ fn test_learner_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
 
     // Remove learner (4, 10) from region 1.
     pd_client.must_remove_peer(r1, new_learner_peer(4, 10));
+    must_region_state(&engine_4, r1, PeerState::Tombstone);
```
Comment (BusyJay) [inline tests/raftstore_cases/test_conf_change.rs]: Is it the same as `must_get_none(&engine_4, b"k2")`?

**Comment by Hoverbear** [comment]
I've been seeing this bug more often with #3289, this may be related/

File: tests/raftstore_cases/test_conf_change.rs [hunk 2]
```diff
@@ -131,11 +132,28 @@ fn test_simple_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
...
+    let mut rs = get_region_state(engine_2.as_ref(), r1);
+    rs.mut_region().mut_region_epoch().conf_ver += 100;
+    put_region_state(engine_2.as_ref(), r1, &rs);
 
     // add peer (2, 4) to region 1.
     pd_client.must_add_peer(r1, new_peer(2, 4));
+    let mut peer_ok = false;
+    for _ in 0..30 {
+        thread::sleep(Duration::from_millis(20));
+        if get_region_state(&engine_2, r1).get_state() != PeerState::Normal {
```
Comment (siddontang) [inline tests/raftstore_cases/test_conf_change.rs]: seem the state must be Tombstone here?

File: tests/raftstore_cases/test_conf_change.rs [hunk 3]
```diff
@@ -131,11 +132,28 @@ fn test_simple_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
...
+            continue;
+        }
+        peer_ok = true;
```
Comment (siddontang) [inline tests/raftstore_cases/test_conf_change.rs]: maybe we can do the assert here directly

File: tests/raftstore_cases/test_conf_change.rs [hunk 4]
```diff
@@ -131,11 +132,28 @@ fn test_simple_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
...
+    }
+    assert!(!peer_ok);
 
     // Remove peer (3, 3) from region 1.
+    rs.mut_region().mut_region_epoch().conf_ver -= 100;
```
Comment (siddontang) [inline tests/raftstore_cases/test_conf_change.rs]: I think we need to check epoch conf_ver with the same value before

File: tests/raftstore_cases/test_conf_change.rs [hunk 5]
```diff
@@ -610,8 +628,17 @@ fn test_learner_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
     must_get_equal(&engine_4, b"k1", b"v1");
     must_get_equal(&engine_4, b"k2", b"v2");
 
+    // Can't add duplicate learner.
+    let epoch1 = pd_client.get_region_epoch(r1);
+    pd_client.add_peer(r1, new_learner_peer(4, 10));
+    pd_client.must_add_peer(r1, new_peer(5, 100)); // For ensure the last is proposed.
```
Comment (BusyJay) [inline tests/raftstore_cases/test_conf_change.rs]: Why can this ensure the learner is proposed?

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @Hoverbear

**Comment by hicqu** [comment]
ping @BusyJay @Hoverbear .

**Comment by hicqu** [comment]
ping @BusyJay .

**Pull request merged**